File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        indices = {}
        for i in range(len(nums)):
            required = target - nums[i]
            if required in indices:
                return i, indices[required]
            indices[nums[i]] = i
            

File: 1-two-sum/1-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> map = new Dictionary<int, int>();
        int remaining;
        for (int i = 0; i < nums.Length; i++) {
            remaining = target - nums[i];
            if (map.ContainsKey(remaining)) {
                return new int[] {i, map[remaining]};
            }
            map[nums[i]] = i;
        }
        return new int[0];
    }
}
